homeassistant/components/aquacell/sensor.py

```diff
@@ -0,0 +1,157 @@
+"""Sensors exposing properties of the softener device."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime
+from typing import cast
+
+from aioaquacell import Softener
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfTime
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AquacellCoordinator
+from .entity import AquacellEntity
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass(frozen=True, kw_only=True)
+class SoftenerEntityDescription(SensorEntityDescription):
+    """Describes Softener sensor entity."""
+
+    value_fn: Callable[[Softener], str | int | float | None]
+
+
+SENSORS: tuple[SoftenerEntityDescription, ...] = (
+    SoftenerEntityDescription(
+        key="salt_leftpercentage",
+        translation_key="salt_leftpercentage",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:basket-fill",
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda softener: cast(int, softener.salt.leftPercent),
```

Comment:
Does aioaquacell contain enough typing? Consider adding a `py.typed` file to enable mypy to read the `aioaquacell` types